llmfoundry/models/mpt/modeling_mpt.py

```diff
@@ -437,7 +442,7 @@ def construct_blocks(self, config: MPTConfig) -> nn.ModuleList:
         block_args = self.extract_block_args(config.to_dict())
 
         return nn.ModuleList([
-            MPTBlock(
+            self.block_class(
```

Comment:
thanks for adding this -- unit scaling probably will need to use this for example. Would it make sense to have a block registry?